### PR TITLE
warn when profiling.ignore_first_call isn't used.

### PR DESCRIPTION
**Pull request by nouiz**
This will help user find the strange profiling information themself without asking questions.

### CODE REVIEW

File: theano/compile/profiling.py [hunk 1]
```diff
@@ -206,9 +210,12 @@ def reset(self):
...
     def __init__(self, atexit_print=True, flag_time_thunks=None, **kwargs):
-        if (hasattr(theano, 'sandbox') and
-                hasattr(theano.sandbox, 'cuda') and
-                theano.sandbox.cuda.cuda_enabled):
+        if (config.profile and
+                ((hasattr(theano, 'sandbox') and
+                  hasattr(theano.sandbox, 'cuda') and
+                  theano.sandbox.cuda.cuda_enabled)) or(
+                      hasattr(theano, 'gpuarray') and
+                      theano.gpuarray.pygpu_activated)):
```
Comment (abergeron) [inline theano/compile/profiling.py]: This or clause is outside of the config.profile one which causes the error to happen all the time.

Comment (nouiz) [inline theano/compile/profiling.py]: fixed

File: theano/gof/vm.py [hunk 2]
```diff
@@ -752,9 +752,11 @@ def accept(self, fgraph, no_recycling=None, profile=None):
 
         """
         if (config.profile and
-                hasattr(theano, 'sandbox') and
-                hasattr(theano.sandbox, 'cuda') and
-                theano.sandbox.cuda.cuda_enabled):
+                ((hasattr(theano, 'sandbox') and
+                  hasattr(theano.sandbox, 'cuda') and
+                  theano.sandbox.cuda.cuda_enabled) or(
```
Comment (abergeron) [inline theano/gof/vm.py]: The parenthesis should be on the next line.

Comment (abergeron) [inline theano/gof/vm.py]: Or at least have a space before.

Comment (nouiz) [inline theano/gof/vm.py]: fixed

**Pull request merged**